cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/DeprecateCommand.java

```diff
@@ -0,0 +1,201 @@
+        if (!validateOrgName(packageSignature.split("/")[0])) {
+            CommandUtil.printError(errStream, "invalid package organization. Only alphanumerics and underscores" +
+                            " are allowed in organization name. Provide a valid package organization. ",
+                    USAGE_TEXT, false);
+            CommandUtil.exitError(this.exitWhenFinish);
+            return;
+        }
+        if (!validatePackageName(packageSignature.split("/")[1])) {
+            CommandUtil.printError(errStream, "invalid package name. Only alphanumerics, underscores and " +
+                            "periods are allowed in a package name and the maximum length is 256 characters. " +
+                            "Provide a valid package name.",
+                    USAGE_TEXT, false);
+            CommandUtil.exitError(this.exitWhenFinish);
+            return;
+        }
+        if (!validatePackageVersion(packageValue.split(":")[1])) {
+            CommandUtil.printError(errStream, "invalid package. Provide a valid package version.",
+                    USAGE_TEXT, false);
+            CommandUtil.exitError(this.exitWhenFinish);
+            return;
+        }
+        // Skip --includes flag if it is set without code coverage
+        if (deprecationMsg != null && undoFlag) {
+            this.outStream.println("warning: ignoring --message (-m) flag since this is an undo request");
+        }
+        deprecateInCentral(packageValue);
+
+        if (this.exitWhenFinish) {
+            Runtime.getRuntime().exit(0);
+        }
+    }
+
+    @Override
+    public String getName() {
+        return DEPRECATE_COMMAND;
+    }
+
+    @Override
+    public void printLongDesc(StringBuilder out) {
+        out.append("deprecates a package in Ballerina Central \n");
+    }
+
+    @Override
+    public void printUsage(StringBuilder out) {
+        out.append(" bal deprecate {<org-name>/<package-name>:<version>} \n");
+    }
+
+    @Override
+    public void setParentCmdParser(CommandLine parentCmdParser) {
+    }
+
+    /**
+     * Deprecate a package in central.
+     *
+     * @param packageInfo package to deprecate.
+     */
+    private void deprecateInCentral(String packageInfo) {
+        try {
+            Settings settings;
+            try {
+                settings = RepoUtils.readSettings();
+                // Ignore Settings.toml diagnostics in the deprecate command
+            } catch (SettingsTomlException e) {
+                // Ignore 'Settings.toml' parsing errors and return empty Settings object
+                settings = Settings.from();
+            }
+            CentralAPIClient client = new CentralAPIClient(RepoUtils.getRemoteRepoURL(),
+                    initializeProxy(settings.getProxy()),
+                    getAccessTokenOfCLI(settings));
+            client.deprecatePackage(packageInfo, deprecationMsg,
+                    JvmTarget.JAVA_11.code(),
```

Comment:
Is this version related to the Central or is it the platform version of the package that is deprecated? If it is the latter, then we need to support `any` as well.

Reply:
This version is related to the Central request.